homeassistant/helpers/config_validation.py

```diff
@@ -361,6 +361,22 @@ def enum(enumClass: type[Enum]) -> vol.All:
     return vol.All(vol.In(enumClass.__members__), enumClass.__getitem__)
 
 
+def enum_value(enumClass: type[Enum]) -> Callable[[Any], Enum]:
```

Comment:
So the validator looks ok, but I am thinking here: isn't this a footgun because shouldn't the value of an enum not matter?

I guess in the entity category case it's used for an external API so we should never change it anymore.